library/types/test/string_test.rb

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+Yast.import "String"
+
+describe Yast::String do
+  before do
+    # ensure proper default locale
+    ENV["LANG"] = "C"
+  end
+
+  describe ".Quote" do
+    it "returns empty string if nil passed" do
+      expect(subject.Quote(nil)).to eq ""
+    end
+
+    it "returns all single quotes escaped" do
+      expect(subject.Quote("")).to eq ""
+      expect(subject.Quote("a")).to eq "a"
+      expect(subject.Quote("a'b")).to eq "a'\\''b"
+      expect(subject.Quote("a'b'c")).to eq "a'\\''b'\\''c"
+    end
+  end
+
+  describe ".UnQuote" do
+    it "returns empty string if nil passed" do
+      expect(subject.UnQuote(nil)).to eq ""
+    end
+
+    it "returns unescaped single quotes" do
+      expect(subject.UnQuote("")).to eq ""
+      expect(subject.UnQuote("a")).to eq "a"
+      expect(subject.UnQuote("a'\\''b")).to eq "a'b"
+      expect(subject.UnQuote("a'\\''b'\\''c")).to eq "a'b'c"
+      expect(subject.UnQuote("a'\\'''\\''b'\\''c")).to eq "a''b'c"
+    end
+  end
+
+  describe ".FormatSize" do
+    it "returns empty string if nil passed" do
+      expect(subject.FormatSize(nil)).to eq ""
+    end
+
+    FORMAT_SIZE_DATA = {
```

Comment:
maybe old habit to use constants for constant data :) No strong opinion for any solution.